src/rpc/RPCHelpers.cpp

```diff
@@ -1036,6 +1053,91 @@ accountHolds(
     return amount;
 }
 
+ripple::STAmount
+accountHolds(
+    BackendInterface const& backend,
+    data::AmendmentCenterInterface const& amendmentCenter,
+    std::uint32_t sequence,
+    ripple::AccountID const& account,
+    ripple::Currency const& currency,
+    ripple::AccountID const& issuer,
+    bool const zeroIfFrozen,
+    boost::asio::yield_context yield
+)
+{
+    ripple::STAmount amount;
+    if (ripple::isXRP(currency))
+        return {xrpLiquid(backend, sequence, account, yield)};
+
+    auto const key = ripple::keylet::line(account, issuer, currency).key;
+    auto const blob = backend.fetchLedgerObject(key, sequence, yield);
+
+    if (!blob) {
+        amount.setIssue(ripple::Issue(currency, issuer));
+        amount.clear();
+        return amount;
+    }
+
+    ripple::SerialIter it{blob->data(), blob->size()};
+    ripple::SLE const sle{it, key};
```

Comment:
You don't need it if `allowBalance` is false.

Reply:
fixed